### ISSUE TITLE
Proxy models and environment file

### ISSUE THREAD

**Issue by harskish** [yuke93/ReconfigISP#2]
Hi!

I would like to make use of this project in my research, but unfortunately that seems hard in its current state.
Specifically, I would like to run (and potentially train) proxy models. However, the repo does not seem to contain any pretrained models, or instructions outlining how to train new proxies from scratch.

Additionally, it is not currently clear where the reference non-differentiable origin models are located. [tools_origin.py](https://github.com/yuke93/ReconfigISP/blob/main/codes/models/modules/tools_origin.py) contains the following imports:
```
import whitebalance as wb
import gamma as gm
import demosaic as dm
import globaltonemapping as gtm
import spatialnoisereduction as snr
```

However, since there is no environment.yml or requirements.txt, it is hard to know whether these files are local to the project and not included in the repo, or 3rd party pip/conda packages.

Would you consider adding an environment file and/or the missing origin models? Instructions related to the training of proxy models (and pretrained proxy models, if possible) would also be very helpful.

Thank you!
-Erik

**Comment by NeoForNew** [comment]
I have encountered the same questions.

**Comment by NeoForNew** [comment]
This is unofficial env list:
`name: reconfig
channels:
  - nvidia
  - conda-forge
  - pytorch
dependencies:
  - _libgcc_mutex=0.1=conda_forge
  - _openmp_mutex=4.5=2_kmp_llvm
  - blas=2.116=mkl
  - blas-devel=3.9.0=16_linux64_mkl
  - brotlipy=0.7.0=py39hb9d737c_1005
  - bzip2=1.0.8=h7f98852_4
  - ca-certificates=2022.12.7=ha878542_0
  - certifi=2022.12.7=pyhd8ed1ab_0
  - cffi=1.14.6=py39he32792d_0
  - charset-normalizer=2.1.1=pyhd8ed1ab_0
  - cryptography=39.0.0=py39hd598818_0
  - cuda=11.7.1=0
  - cuda-cccl=11.7.91=0
  - cuda-command-line-tools=11.7.1=0
  - cuda-compiler=11.7.1=0
  - cuda-cudart=11.7.99=0
  - cuda-cudart-dev=11.7.99=0
  - cuda-cuobjdump=11.7.91=0
  - cuda-cupti=11.7.101=0
  - cuda-cuxxfilt=11.7.91=0
  - cuda-demo-suite=12.0.140=0
  - cuda-documentation=12.0.140=0
  - cuda-driver-dev=11.7.99=0
  - cuda-gdb=12.0.140=0
  - cuda-libraries=11.7.1=0
  - cuda-libraries-dev=11.7.1=0
  - cuda-memcheck=11.8.86=0
  - cuda-nsight=12.0.140=0
  - cuda-nsight-compute=12.0.1=0
  - cuda-nvcc=11.7.99=0
  - cuda-nvdisasm=12.0.140=0
  - cuda-nvml-dev=11.7.91=0
  - cuda-nvprof=12.0.146=0
  - cuda-nvprune=11.7.91=0
  - cuda-nvrtc=11.7.99=0
  - cuda-nvrtc-dev=11.7.99=0
  - cuda-nvtx=11.7.91=0
  - cuda-nvvp=12.0.146=0
  - cuda-runtime=11.7.1=0
  - cuda-sanitizer-api=12.0.140=0
  - cuda-toolkit=11.7.1=0
  - cuda-tools=11.7.1=0
  - cuda-visual-tools=11.7.1=0
  - ffmpeg=4.3=hf484d3e_0
  - freetype=2.12.1=hca18f0e_1
  - gds-tools=1.5.1.14=0
  - gmp=6.2.1=h58526e2_0
  - gnutls=3.6.13=h85f3911_1
  - icu=70.1=h27087fc_0
  - idna=3.4=pyhd8ed1ab_0
  - jpeg=9e=h0b41bf4_3
  - lame=3.100=h166bdaf_1003
  - lcms2=2.14=hfd0df8a_1
  - ld_impl_linux-64=2.40=h41732ed_0
  - lerc=4.0.0=h27087fc_0
  - libblas=3.9.0=16_linux64_mkl
  - libcblas=3.9.0=16_linux64_mkl
  - libcublas=11.10.3.66=0
  - libcublas-dev=11.10.3.66=0
  - libcufft=10.7.2.124=h4fbf590_0
  - libcufft-dev=10.7.2.124=h98a8f43_0
  - libcufile=1.5.1.14=0
  - libcufile-dev=1.5.1.14=0
  - libcurand=10.3.1.124=0
  - libcurand-dev=10.3.1.124=0
  - libcusolver=11.4.0.1=0
  - libcusolver-dev=11.4.0.1=0
  - libcusparse=11.7.4.91=0
  - libcusparse-dev=11.7.4.91=0
  - libdeflate=1.17=h0b41bf4_0
  - libffi=3.3=h58526e2_2
  - libgcc-ng=12.2.0=h65d4601_19
  - libgfortran-ng=12.2.0=h69a702a_19
  - libgfortran5=12.2.0=h337968e_19
  - libgomp=12.2.0=h65d4601_19
  - libhwloc=2.8.0=h32351e8_1
  - libiconv=1.17=h166bdaf_0
  - liblapack=3.9.0=16_linux64_mkl
  - liblapacke=3.9.0=16_linux64_mkl
  - libnpp=11.7.4.75=0
  - libnpp-dev=11.7.4.75=0
  - libnvjpeg=11.8.0.2=0
  - libnvjpeg-dev=11.8.0.2=0
  - libpng=1.6.39=h753d276_0
  - libsqlite=3.40.0=h753d276_0
  - libstdcxx-ng=12.2.0=h46fd767_19
  - libtiff=4.5.0=h6adf6a1_2
  - libwebp-base=1.2.4=h166bdaf_0
  - libxcb=1.13=h7f98852_1004
  - libxml2=2.10.3=h7463322_0
  - libzlib=1.2.13=h166bdaf_4
  - llvm-openmp=15.0.7=h0cdce71_0
  - mkl=2022.1.0=h84fe81f_915
  - mkl-devel=2022.1.0=ha770c72_916
  - mkl-include=2022.1.0=h84fe81f_915
  - ncurses=6.3=h27087fc_1
  - nettle=3.6=he412f7d_0
  - nsight-compute=2022.4.1.6=0
  - numpy=1.24.2=py39h7360e5f_0
  - openh264=2.1.1=h780b84a_0
  - openjpeg=2.5.0=hfec8fc6_2
  - openssl=1.1.1t=h0b41bf4_0
  - pillow=9.4.0=py39h2320bf1_1
  - pip=23.0=pyhd8ed1ab_0
  - pthread-stubs=0.4=h36c2ea0_1001
  - pycparser=2.21=pyhd8ed1ab_0
  - pyopenssl=23.0.0=pyhd8ed1ab_0
  - pysocks=1.7.1=pyha2e5f31_6
  - python=3.9.0=hffdb5ce_5_cpython
  - python_abi=3.9=3_cp39
  - pytorch=1.13.1=py3.9_cuda11.7_cudnn8.5.0_0
  - pytorch-cuda=11.7=h67b0de4_1
  - pytorch-mutex=1.0=cuda
  - readline=8.1.2=h0f457ee_0
  - requests=2.28.2=pyhd8ed1ab_0
  - setuptools=67.3.1=pyhd8ed1ab_0
  - sqlite=3.40.0=h4ff8645_0
  - tbb=2021.7.0=h924138e_1
  - tk=8.6.12=h27826a3_0
  - torchaudio=0.13.1=py39_cu117
  - torchvision=0.14.1=py39_cu117
  - typing_extensions=4.4.0=pyha770c72_0
  - tzdata=2022g=h191b570_0
  - urllib3=1.26.14=pyhd8ed1ab_0
  - wheel=0.38.4=pyhd8ed1ab_0
  - xorg-libxau=1.0.9=h7f98852_0
  - xorg-libxdmcp=1.1.3=h7f98852_0
  - xz=5.2.6=h166bdaf_0
  - zlib=1.2.13=h166bdaf_4
  - zstd=1.5.2=h3eb15da_6
  - pip:
    - absl-py==1.4.0
    - cachetools==5.3.0
    - contourpy==1.0.7
    - cycler==0.11.0
    - fonttools==4.38.0
    - google-auth==2.16.0
    - google-auth-oauthlib==0.4.6
    - grpcio==1.51.1
    - importlib-metadata==6.0.0
    - importlib-resources==5.10.2
    - kiwisolver==1.4.4
    - lmdb==1.4.0
    - markdown==3.4.1
    - markupsafe==2.1.2
    - matplotlib==3.7.0
    - oauthlib==3.2.2
    - opencv-python==4.7.0.68
    - packaging==23.0
    - protobuf==4.21.12
    - pyasn1==0.4.8
    - pyasn1-modules==0.2.8
    - pyparsing==3.0.9
    - python-dateutil==2.8.2
    - pyyaml==6.0
    - requests-oauthlib==1.3.1
    - rsa==4.9
    - six==1.16.0
    - tensorboard==2.12.0
    - tensorboard-data-server==0.7.0
    - tensorboard-plugin-wit==1.8.1
    - tqdm==4.64.1
    - werkzeug==2.2.3
    - zipp==3.13.0
prefix: /opt/conda/envs/reconfig
`

**Comment by cuiziteng** [comment]
I have the same issue
